### PR TITLE
Update the heading block to use the colors hook

### PR DESCRIPTION
**Pull request by youknowriad**
Based on #21021 to try test the API on different blocks (heading for this PR)

It also adds background color support to the heading block. At the moment there's no way to choose just text color, but it's easy to implement if needed. I just thought being consistent with other blocks is a good thing.

### CODE REVIEW

File: packages/block-library/src/heading/style.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+h1,
```
Comment (gziolo) [inline packages/block-library/src/heading/style.scss]: The same question as for Paragraph block:

Are you sure it’s safe to set those values for all headings including front-end as part of the block definition?

**Review by gziolo — COMMENTED** [review]
It’s must have to hide __experimentalUseColors as in implementation detail as it’s hard to follow as you need to know all those differences between elements and components exposed and hoe to wire them. Did you find any limitations of the support flag?

**Comment by youknowriad** [comment]
> Did you find any limitations of the support flag?

The biggest limit is that you can only apply these styles/classnames to the block wrapper. I think it's probably enough for 90% of the cases.

**Comment by gziolo** [comment]
It needs to be rebased with master. I see changes from the parent PR in here that makes the final call harder

**Comment by youknowriad** [comment]
@gziolo it should be better now, the base branch was wrong.

**Review by gziolo — APPROVED** [review]
Overall it's good to go. Nice work!

Depending on how the proposal from @nosolosw shared in https://github.com/WordPress/gutenberg/pull/21037#discussion_r397068844 ends up implemented, CSS code added in this PR should be updated. I guess it's fine to merge and update later.

**Comment by paaljoachim** [comment]
Is Heading background color UI being added or is it just that it is implemented but not added through the UI? It would be good to add the Heading background UI control while your first working with this...:)

**Comment by youknowriad** [comment]
it's added in the UI

File: packages/block-library/src/heading/deprecated.js [hunk 2]
```diff
@@ -30,17 +31,77 @@ const blockAttributes = {
...
+	return {
+		...omit( attributes, [ 'customTextColor' ] ),
+		style,
+	};
 };
 
 const deprecated = [
 	{
-		attributes: blockAttributes,
+		supports: blockSupports,
```
Comment (gziolo) [inline packages/block-library/src/heading/deprecated.js]: I forgot about one thing, it would be nice to add the test case that covers this new deprecation:
- a block that has `has-text-color` class set?

@jorgefilipecosta should know best what's the snippet that makes this deprecation unique.

Comment (jorgefilipecosta) [reply packages/block-library/src/heading/deprecated.js]: I will look into the deprecations and try to add a snippet specific for this deprecation 👍

Comment (gziolo) [reply packages/block-library/src/heading/deprecated.js]: Thanks @jorgefilipecosta 🙇
